--- ../src/gtk/window.cpp
+++ ../src/gtk/window.cpp
@@ -775,33 +775,6 @@ wxTranslateGTKKeyEventToWx(wxKeyEvent& event,
                 keysym = (KeySym)gdk_event->string[0];
             }
 
-#ifdef GDK_WINDOWING_X11
-            // we want to always get the same key code when the same key is
-            // pressed regardless of the state of the modifiers, i.e. on a
-            // standard US keyboard pressing '5' or '%' ('5' key with
-            // Shift) should result in the same key code in OnKeyDown():
-            // '5' (although OnChar() will get either '5' or '%').
-            //
-            // to do it we first translate keysym to keycode (== scan code)
-            // and then back but always using the lower register
-            Display *dpy = (Display *)wxGetDisplay();
-            KeyCode keycode = XKeysymToKeycode(dpy, keysym);
-
-            wxLogTrace(TRACE_KEYS, wxT("\t-> keycode %d"), keycode);
-
-#ifdef HAVE_X11_XKBLIB_H
-            KeySym keysymNormalized = XkbKeycodeToKeysym(dpy, keycode, 0, 0);
-#else
-            KeySym keysymNormalized = XKeycodeToKeysym(dpy, keycode, 0);
-#endif
-
-            // use the normalized, i.e. lower register, keysym if we've
-            // got one
-            key_code = keysymNormalized ? keysymNormalized : keysym;
-#else
-            key_code = keysym;
-#endif
-
             // as explained above, we want to have lower register key codes
             // normally but for the letter keys we want to have the upper ones
             //
